src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newStates = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'addProperties':
+        const StateBfAdd = { ...(newStates[newStates.length - 1] || state) };
+
+        newStates.push({ ...StateBfAdd, ...action.extraData });
+        break;
+      case 'removeProperties':
+        const stateBfRemove = { ...(newStates[newStates.length - 1] || state) };
+
+        action.keysToRemove.forEach((key) => {
+          if (Object.keys(stateBfRemove).includes(key)) {
+            delete stateBfRemove[key];
```

Comment:
[CODE STYLE] - Checking if the key exists in the object before deleting is unnecessary because the delete operator will not throw an error if the property does not exist. You can remove the 'if' check to simplify the code.